### PR TITLE
feat: require type on Token interface

### PR DESCRIPTION
**Pull request by koalaty-code**
## Description

- Marks `type` as required on the Token interface

## Testing required outside of automated testing?

- [X] Not Applicable

### Screenshots (if appropriate):

- [X] Not Applicable

## Rollback / Rollforward Procedure

- [X] Roll Forward
- [ ] Roll Back

## Reviewer Checklist

- [ ] Description of Change
- [ ] Description of outside testing if applicable.
- [ ] Description of Roll Forward / Backward Procedure
- [ ] Documentation updated for Change

### CODE REVIEW

File: library/test/tokens.test.ts [hunk 1]
```diff
@@ -764,6 +765,7 @@ describe('Tokens', () => {
     it('should reject with status >= 400 <= 599', async () => {
       const parentId = chance.string();
       const tokenPayload = {
+        type: 'token' as TokenType,
```
Comment (djejaquino) [inline library/test/tokens.test.ts:768]: Do you face any errors without this cast?

Comment (koalaty-code) [reply library/test/tokens.test.ts:768]: Yeah, I couldn't figure that out.. 
image

File: library/src/tokens/types.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ interface TokenEncryption {
 interface Token {
   id: string;
   tenantId: string;
-  type?: TokenType;
+  type: TokenType;
```
Comment (koalaty-code) [inline library/src/tokens/types.ts:25]: @brigonzalez @lcschy @djejaquino is there a way to have a default value here? We currently default to `"token"` in the dotnet SDK, but I couldn't find a way to supply default values with TS interfaces/types.

Comment (djejaquino) [reply library/src/tokens/types.ts:25]: You can't set a default value in interfaces. You could make it optional in the `CreateTokenModel` and set the default value in the implementation.

**Comment by djejaquino** [comment]
> > Also, deprecated `createToken` method should be removed.
> 
> @djejaquino would you like to remove all the deprecated token methods? `createToken`, `getToken`, and `deleteToken`?

Yes, please. I don't see them being used anywhere.

**Pull request merged**